hayeon/baekjoon/수학/11050-이항-계수-1.js

```diff
@@ -0,0 +1,40 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const setFactorial = (num) => {
+    let factorial = 1;
+
+    for (let i = 1; i <= num; i++) {
+      factorial *= i;
+    }
+    return factorial;
+  };
+
+  const [naturalNum, integer] = input[0].split(" ").map(Number);
+  let result = 0;
+
+  const numerator = setFactorial(naturalNum);
+  const denominator =
+    setFactorial(integer) * setFactorial(naturalNum - integer);
+
+  result = Math.floor(numerator / denominator);
+
+  console.log(result);
```

Comment:
```suggestion
  const getFactorial = (num) => {
    let factorial = 1;

    for (let i = 1; i <= num; i++) {
      factorial *= i;
    }
    return factorial;
  };

  const [naturalNum, integer] = input[0].split(" ").map(Number);
  let result = 0;

  const numerator = getFactorial(naturalNum);
  const denominator =
    getFactorial(integer) * getFactorial(naturalNum - integer);

  result = Math.floor(numerator / denominator);

  console.log(result);
```
- 풀이 잘 봤습니다 🙇‍♂️
- 그리고 set은 상태를 셋팅한다는 의미인데 현재는 factorial값을 가져와서 특정 변수에 할당을 해주고 있기 때문에 set보다는 get이 더 올바른 네이밍입니다!
- 아래와 같이 반복문이 아닌 재귀함수로 팩토리얼을 구해낼 수도 있습니다.
```js
const getFactorial = (num) => {
  if (num < 2) return 1;
  return num * factorial(num - 1);
};
```

Reply:
오호 그렇군요!!! 네이밍 더 신경쓰겟습니닷!! 재귀함수로 팩토리얼 구하는법도 처음 알았네요.....!!